templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -385,8 +385,23 @@
           function checkEmail() {
             var data = $("#contact_id", $form).select2('data');
             if (data && data.extra && data.extra.email && data.extra.email.length) {
-              $("#email-receipt", $form).show();
-              $("#email-address", $form).html(data.extra.email);
+              CRM.api4('Email', 'get', {
+                select: ["on_hold"],
+                where: [["contact_id", "=", data.id], ["is_primary", "=", true]],
+                chain: {"contact_do_not_email":["Contact", "get", {"where":[["id", "=", data.id]], "select":["do_not_email"]}]}
```

Comment:
Not a blocker since it doesn't make too much practical difference, but a join might be better here, since if they have e.g. 5 email addresses this creates a total of 6 api calls.

Reply:
I think it's just two because we are only getting primary email, but point taken, have changed to join.

Reply:
ah right.